File: packages/react-native-nitro-image/android/src/main/java/com/margelo/nitro/image/HybridTestView.kt
package com.margelo.nitro.image

import android.graphics.Color
import android.view.View
import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.facebook.react.uimanager.ThemedReactContext

@Keep
@DoNotStrip
class HybridTestView(val context: ThemedReactContext): HybridTestViewSpec() {
    // View
    override val view: View = View(context)

    // Props
    private var _isBlue = false
    override var isBlue: Boolean
        get() = _isBlue
        set(value) {
            _isBlue = value
            val color = if (value) Color.BLUE else Color.RED
            view.setBackgroundColor(color)
        }
    override colorScheme: ColorScheme = ColorScheme.LIGHT
    override var someCallback: () -> Unit = {}

    // Methods
    override fun someMethod(): Unit {
        someCallback()
    }
}

File: packages/react-native-nitro-image/ios/HybridTestView.swift
//
//  HybridTestView.swift
//  react-native-nitro-image
//
//  Created by Marc Rousavy on 03.10.24.
//

import Foundation
import UIKit

class HybridTestView : HybridTestViewSpec {
  // UIView
  var view: UIView = UIView()

  // Props
  var isBlue: Bool = false {
    didSet {
      view.backgroundColor = isBlue ? .systemBlue : .systemRed
    }
  }
}
